### PR TITLE
Add `--cvss-fail-threshold`

### LINKED ISSUE #114: Change from fail-on-result to fail-on-cvss
The [dependency-check Maven plugin](https://jeremylong.github.io/DependencyCheck/dependency-check-maven/check-mojo.html) has deprecated the `failBuildOnAnyVulnerability` in favor of setting `failBuildOnCVSS` to 0 for all vulnerabilities.

This allows for greater flexibility and granularity, making CI adoption more attractive. For example, a team could opt to fail builds with any vulnerabilities with a CVSS > 6. This can help reduce churn a little, as lower-criticality issues can be addressed at more leisure.

### PR DESCRIPTION
**Pull request by lread**
See README for general description.

New option is mutually exclusive to `--fail-on-result`; if both are specified, clj-watson fails fast with usage error and help.

Conservatively derives score when missing or suspicious looking:
- When severity is available, conservatively converts to score
- Since we don't know if if score is CVSS2 or CVSS3/4 derives, High and Critical to 10.0, Medium and Low are converted to upper bound of their ranges.
- The experimental github-advisory strategy seems to regularly populate score with `0.0` but with a valid looking severity; we treat a score of 0.0 as suspicious.
- I've not seen cases of invalid severities in the wild, but we handle them just the same, when we can't make sense of things we derive to the most critical score which is 10.0.

Also:
- factored out table support from cli-spec ns to new table ns to reuse it for summary table.
- renamed summarize fn to final-summary to better distinguish from our new summary fn
- a new utils ns for `assoc-some` fn (cribbed clj-kondo which cribbed from medley).

Closes #114

### CODE REVIEW

File: src/clj_watson/cli_spec.clj [hunk 1]
```diff
@@ -277,7 +244,15 @@
 
       :else
       (let [opts (cli/parse-opts orig-args {:spec spec-scan-args :error-fn usage-error :restrict true})]
-        (report-deprecations opts)
+        (if (and (:cvss-fail-threshold opts) (:fail-on-result opts))
+          (usage-error {:type :clj-watson/cli
+                        :msg (format "Invalid usage, specificy only one of: %s"
```
[the remark this answers is not in the document]
Comment (lread) [reply src/clj_watson/cli_spec.clj:249]: Thanks!

**Comment by lread** [comment]
Thanks for taking a peek @coyotesqrl.
I think this matches what we discussed.
If you have any other feedback, happy to hear it!

**Comment by lread** [comment]
@seancorfield when you have some time and some interest, this PR awaits your review.

**Comment by seancorfield** [comment]
Sorry, last week got completely derailed which was when I said I would review this...

**Comment by lread** [comment]
Thanks! And no problem @seancorfield, I understand that folks get busy and distracted!

**Pull request merged**